docs/naps/8-telemetry.md

```diff
@@ -0,0 +1,182 @@
+:orphan: 
+  
+ (nap-8)= 
+  
+ # NAP-8 — Telemetry
+  
+ ```{eval-rst} 
+ :Author: Grzegorz Bokota 
+ :Created: <date created on, in yyyy-mm-dd format> 
+ :Resolution: <url> (required for Accepted | Rejected | Withdrawn) 
+ :Resolved: <date resolved, in yyyy-mm-dd format> 
+ :Status: Draft
+ :Type: Standards Track 
+ :Version effective: <version-number> (for accepted NAPs) 
+ ``` 
+  
+ ## Abstract 
+  
+ This NAP is describes why telemetry is helpful to the napari project and the architecture and solutions selected to maximize the privacy of our users. 
+ 
+ ## Motivation and Scope
+
+With the growth of napari, the standard feedback loop through napari community meetings and napari-related events at conferences has reached its capacity. Also, we collect many feature requests for which we cannot find volunteers for implementation. 
+
+To have the possibility of sustainable development of the project we will either need to rely on paid contractors or on companies donating employee time managed by the core devs.
+
+Both scenarios require us to provide some information about the estimated number of users to prove to potential funders that their donation/grant will be used in a valuable way. 
+
+Adding the option for monitoring plugin usage allows us to identify the most important plugins and try to establish cooperation with their maintainers to reduce the probability that the plugin will not be ready for a new napari release. Such monitoring could contain not only the list of installed plugins but also which commands and contributions are used most often.
+
+Also collecting information about data types and their size will provide valuable information about the typical use cases of napari.
+
+Still, users need to be able to opt out of such monitoring, and adjust the level of detail of the information that is sent to the napari server.
```

Comment:
Detail, I would store the date when someone agreed, and maybe reconfirm every few month or so.

You might need a way for the remote to reply with a specific error code/version number that void the user consent, and retrigger asking consent as well.

Reply:
Why do you prefer longer storage?

Reply:
I think the proposal is that we reconfirm opt-in every few months or so. I think this is a good idea and worth adding.

Reply:
I still do not follow. 
Did we want to reconfirm every three months (for example), or when we update the telemetry code, so change the amount of collected information?

Reply:
I think it should be `min(3_months, package_update)` for re-confirming, personally. 

The dialog for whether you want to continue sending telemetry could show an example of a recent report that had been sent, so the user can gauge their participation based on real data. We could always add a `Don't ask me again` option as well of course.